### PR TITLE
[ao][sparsity] adding type hints to sparsifier and utils

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79774
* __->__ #79598
* #79597

Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: torch/ao/sparsity/sparsifier/base_sparsifier.py [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
 __all__ = ["BaseSparsifier"]
 
 SUPPORTED_MODULES = {
     nn.Linear
 }
 
-KEYS_NOT_IN_STATE_DICT = ['module', 'module_fqn', 'tensor_name']
+KEYS_NOT_IN_STATE_DICT = ["module", "module_fqn", "tensor_name"]
+
+__all__ = ["BaseSparsifier"]
```
Comment (z-a-f) [inline torch/ao/sparsity/sparsifier/base_sparsifier.py:24]: dup?

File: torch/ao/sparsity/sparsifier/base_sparsifier.py [hunk 2]
```diff
@@ -42,41 +48,41 @@ class BaseSparsifier(abc.ABC):
...
     """
-    def __init__(self, defaults):
+    def __init__(self, defaults: Optional[Dict[str, Any]] = None):
         super().__init__()
-        self.defaults = defaults
-        if self.defaults is None:
-            self.defaults = dict()
+        if defaults is None:
+            defaults = dict()
+        self.defaults: Dict[str, Any] = defaults
```
Comment (z-a-f) [inline torch/ao/sparsity/sparsifier/base_sparsifier.py:55]: nit: 
```
self.defaults: Dict[str, Any] = defaults or dict()
```

**Review by z-a-f — APPROVED** [review]

File: torch/ao/sparsity/sparsifier/base_sparsifier.py (hunk 1 above)
Comment (HDCharles) [reply torch/ao/sparsity/sparsifier/base_sparsifier.py:24]: pyfmt prefers that i think

File: torch/ao/sparsity/sparsifier/base_sparsifier.py (hunk 2 above)
Comment (HDCharles) [reply torch/ao/sparsity/sparsifier/base_sparsifier.py:55]: that doesn't seem to cause an issue actually

**Pull request closed without merging**